migrate/migrations/000007_add-posts-table.up.sql

```diff
@@ -0,0 +1,31 @@
+CREATE TABLE posts (
+    id INT UNSIGNED AUTO_INCREMENT PRIMARY KEY,
+    created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
+    updated_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
+
+    type TINYINT UNSIGNED NOT NULL CHECK (type IN (0,1,2,3,4)),
+    audience TINYINT UNSIGNED NOT NULL CHECK (audience IN (0,1,2)),
+
+    author_id INT UNSIGNED NOT NULL,
+    title VARCHAR(256) NULL,
+    excerpt VARCHAR(500) NULL,
+    content TEXT NULL,
+    url VARCHAR(1024) NOT NULL,
```

Comment:
I didn't put a `UNIQUE` here to avoid 1. adding a large index and 2. are we 100% sure that URL's will be unique - `ap_id` should be, but it might be possible to have different posts/post_types refer to same URL?

Reply:
Why would we have 2 different posts/post_types referring to same URL? By definition if they are different they should have different URLs right?

Reply:
The `url` property isn't an identifier (https://www.w3.org/TR/activitystreams-vocabulary/#dfn-url)

So I think it's possible that you could have both an `Article` and `Note` with the same `url` representation

Reply:
I might be missing something here, but how can 1 URL resolve to 2 different things? (without additional context)